charts/opentelemetry-collector/templates/_config.tpl

```diff
@@ -193,7 +193,12 @@ receivers:
     exclude: []
     {{- else }}
     # Exclude collector container's logs. The file format is /var/log/pods/<namespace_name>_<pod_name>_<pod_uid>/<container_name>/<run_id>.log
-    exclude: [ /var/log/pods/{{ .Release.Namespace }}_{{ include "opentelemetry-collector.fullname" . }}*_*/{{ include "opentelemetry-collector.lowercase_chartname" . }}/*.log ]
+    exclude:
+    {{- if .Values.namespaceOverride }}
+    - /var/log/pods/{{ .Values.namespaceOverride }}_{{ include "opentelemetry-collector.fullname" . }}*_*/{{ include "opentelemetry-collector.lowercase_chartname" . }}/*.log
+    {{- else }}
+    - /var/log/pods/{{ .Release.Namespace }}_{{ include "opentelemetry-collector.fullname" . }}*_*/{{ include "opentelemetry-collector.lowercase_chartname" . }}/*.log
+    {{- end }}
```

Comment:
instead of an if-else in this template, lets make a new template in `_helpers` that returns `.Values.namespaceOverride` or `.Release.Namespace`.  Then we can do 

```
exclude: [ /var/log/pods/{{ include "opentelemetry-collector.new-template-name }}_{{ include "opentelemetry-collector.fullname" . }}*_*/{{ include "opentelemetry-collector.lowercase_chartname" . }}/*.log ]
```
This keeps the pattern for building the string consistent across all its variable secitons.